src/Helper/Uri.php

```diff
@@ -347,15 +372,22 @@ public function getHost(mixed $default = null): ?StringHelper
 
     /**
      * Returns TRUE if the port is valid.
-     * todo: Implement check for port URI segment
      *
      * @param integer|null $port
      * @return boolean
      */
     public function checkPort(int $port = null): bool
     {
         if ($port === null) {
-            $port = $this->port;
+            $port = $this->getPort();
+        }
+
+        switch ($port) {
```

Comment:
Use `switch (intval($port)) {`

Reply:
But then the tests will fail:
```shell
1) PlanetTeamSpeak\TeamSpeak3Framework\Tests\Helper\UriTest::testCheckUser
Object of class PlanetTeamSpeak\TeamSpeak3Framework\Helper\StringHelper could not be converted to int

C:\Users\Sebastian\Downloads\ts3phpframework\src\Helper\Uri.php:385
C:\Users\Sebastian\Downloads\ts3phpframework\src\Helper\Uri.php:173
C:\Users\Sebastian\Downloads\ts3phpframework\src\Helper\Uri.php:131
C:\Users\Sebastian\Downloads\ts3phpframework\tests\Helper\UriTest.php:87

[+7 similar errors]
```

Reply:
This works:

```php
public function checkPort(int $port = null): bool
{
    if ($port === null) {
        $port = intval($this->port->toString());
    }

    switch ($port) {
        case str_starts_with($port, '-'):
        case $port < 0:
        case !is_int($port):
        case !filter_var($port, FILTER_VALIDATE_INT, ['options' => ['min_range' => 1, 'max_range' => 65535]]):
            return false;
    }

    return true;
}
```

Shall we use this instead?

Reply:
I've pushed it. :)

Reply:
Yes, sorry needs to be string'ed first. My mistake, thanks for fixing!